azure/src/test/java/com/microsoft/azure/management/TestCdn.java

```diff
@@ -0,0 +1,153 @@
+        final String cdnEndpointName = "cdnEndpoint" + this.testId;
+        final String cdnOriginHostName = "mylinuxapp.azurewebsites.net";
+
+        CdnProfile cdnProfile = profiles.define(cdnProfileName)
+                .withRegion(region)
+                .withNewResourceGroup(groupName)
+                .withStandardAkamaiSku()
+                .defineNewEndpoint(cdnEndpointName)
+                    .withOrigin(cdnOriginHostName)
+                    .withGeoFilter("/path/videos", GeoFilterActions.BLOCK, CountryISOCode.ARGENTINA)
+                    .withGeoFilter("/path/images", GeoFilterActions.BLOCK, CountryISOCode.BELGIUM)
+                    .withContentTypeToCompress("text/plain")
+                    .withCompressionEnabled(true)
+                    .withQueryStringCachingBehavior(QueryStringCachingBehavior.BYPASS_CACHING)
+                    .withHttpsAllowed(true)
+                    .withHttpsPort(444)
+                    .withHttpAllowed(false)
+                    .withHttpPort(85)
+                    .attach()
+                .create();
+
+        Assert.assertTrue(cdnProfile.sku().name().equals(SkuName.STANDARD_AKAMAI));
+        Assert.assertNotNull(cdnProfile.endpoints());
+        Assert.assertEquals(1, cdnProfile.endpoints().size());
+        CdnEndpoint endpoint = cdnProfile.endpoints().get(cdnEndpointName);
+        Assert.assertNotNull(endpoint);
+        Assert.assertEquals(cdnOriginHostName, endpoint.originHostName());
+        Assert.assertEquals(444, endpoint.httpsPort());
+        Assert.assertEquals(85, endpoint.httpPort());
+        Assert.assertFalse(endpoint.isHttpAllowed());
+        Assert.assertTrue(endpoint.isHttpsAllowed());
+        Assert.assertTrue(endpoint.isCompressionEnabled());
+        Assert.assertEquals(QueryStringCachingBehavior.BYPASS_CACHING, endpoint.queryStringCachingBehavior());
+        Assert.assertNotNull(endpoint.geoFilters());
+        Assert.assertEquals(QueryStringCachingBehavior.BYPASS_CACHING, endpoint.queryStringCachingBehavior());
+
+        return cdnProfile;
+    }
+
+    @Override
+    public CdnProfile updateResource(CdnProfile profile) throws Exception {
+        String firstEndpointName = profile.endpoints().keySet().iterator().next();
+
+        // Remove an endpoint, update two endpoints and add new one
+        //
+        profile.update()
+                .withTag("provider", "Akamai")
+                .withNewEndpoint("www.bing.com")
+                .defineNewEndpoint("somenewnamefortheendpoint")
+                    .withOrigin("www.contoso.com")
+                    .withGeoFilter("/path/music", GeoFilterActions.BLOCK, CountryISOCode.UNITED_STATES_OUTLYING_ISLANDS)
+                    .attach()
+                .updateEndpoint(firstEndpointName)
+                    .withHttpAllowed(true)
+                    .withHttpPort(1111)
+                    .withoutGeoFilters()
+                    .parent()
+                .apply();
```

Comment:
it is covered here - https://github.com/Azure/azure-sdk-for-java/pull/1288/commits/13db29f01626527162746437ac6f17b9837d83a6#diff-a2dcc8b4d0226a01115fecc7168730a1R142